astropy/io/fits/hdu/compressed/_tiled_compression.py

```diff
@@ -81,6 +82,8 @@ def _header_to_settings(header):
         settings["bytepix"] = 8
         settings["scale"] = int(_get_compression_setting(header, "SCALE", 0))
         settings["smooth"] = _get_compression_setting(header, "SMOOTH", 0)
+    elif compression_type == "JPEGXL":
```

Comment:
I think the biggest issue with this PR is that writing a file like this technically steps on this paragraph from section 10.4 of the FITS 4 spec:

> The name of the permitted algorithms for compressing FITS HDUs, as recorded in the ZCMPTYPE keyword, are listed in Table 36; if other types are later supported, they must be registered with the IAUFWG to reserve the keyword values.

I think it would be a perfectly reasonable policy for astropy to support any registered (even if unspecified) ZCMPTYPE, but I wouldn't want us to write an unregistered key in core.